db_test.go

```diff
@@ -0,0 +1,236 @@
+	Equals(t, port, strconv.Itoa(int(stat.userInput.port)))
+
+	Equals(t, neverSucceedProbe, stat.lastSuccessfulProbe.IsZero())
+	Equals(t, neverFailedProbe, stat.lastUnsuccessfulProbe.IsZero())
+
+	Equals(t, lMin, stat.rttResults.min)
+	Equals(t, lAvg, stat.rttResults.average)
+	Equals(t, lMax, stat.rttResults.max)
+	Equals(t, startTimestamp.Format(timeFormat), stat.startTime.Format(timeFormat))
+	Equals(t, endTimestamp.Format(timeFormat), stat.endTime.Format(timeFormat))
+
+	actualDuration := stat.endTime.Sub(stat.startTime).String()
+	Equals(t, totalDuration, actualDuration)
+	Equals(t, totalUptime, stat.totalUptime.String())
+	Equals(t, totalDowntime, stat.totalDowntime.String())
+	Equals(t, totalPackets, stat.totalSuccessfulProbes+stat.totalUnsuccessfulProbes)
+
+	Equals(t, longestUptime, stat.longestUptime.duration.String())
+	Equals(t, longestUptimeStart.Format(timeFormat), stat.longestUptime.start.Format(timeFormat))
+	Equals(t, longestUptimeEnd.Format(timeFormat), stat.longestUptime.end.Format(timeFormat))
+
+	Equals(t, longestDowntime, stat.longestDowntime.duration.String())
+	Equals(t, longestDowntimeStart.Format(timeFormat), stat.longestDowntime.start.Format(timeFormat))
+	Equals(t, longestDowntimeEnd.Format(timeFormat), stat.longestDowntime.end.Format(timeFormat))
+
+}
+
+func TestSaveHostname(t *testing.T) {
+	// There are many fields, so many things could go wrong; that's why this elaborate test.
+	arg := []string{"localhost", "8001"}
+	s := newDb(arg, ":memory:")
+	defer s.db.Close()
+	stat := mockStats()
+
+	err := s.saveHostNameChange(stat.hostnameChanges)
+	Nil(t, err)
+	// testing the host names if they are properly written
+	query := `SELECT
+	hostname_changed_to, hostname_change_time
+	FROM ` + fmt.Sprintf("%s WHERE event_type IS '%s';", s.tableName, eventTypeHostnameChange)
+
+	rows, err := s.db.Query(query)
+	Nil(t, err)
+
+	idx := 0
+	for rows.Next() {
+		var hostName string
+		var cTime time.Time
+		err = rows.Scan(&hostName, &cTime)
+		Nil(t, err)
+
+		actualHost := stat.hostnameChanges[idx]
+		idx++
+		Equals(t, hostName, actualHost.Addr.String())
+		Equals(t, cTime.Format(timeFormat), actualHost.When.Format(timeFormat))
+	}
+	Equals(t, idx, len(stat.hostnameChanges))
+	rows.Close()
+}
+
+func hostNameChange() []hostnameChange {
+	ipAddresses := []string{
+		"192.168.1.1",
+		"10.0.0.1",
+		"172.16.0.1",
+		"2001:0db8:85a3:0000:0000:8a2e:0370:7334",
+	}
+	var hostNames []hostnameChange
+	for i, ip := range ipAddresses {
+		host := hostnameChange{
+			Addr: netip.MustParseAddr(ip),
+			When: time.Now().Add(time.Duration(i) * time.Minute),
+		}
+		hostNames = append(hostNames, host)
+	}
+	return hostNames
+}
+
+func mockStats() stats {
+	stat := stats{
+		startTime:           time.Now(),
+		endTime:             time.Now().Add(10 * time.Minute),
+		lastSuccessfulProbe: time.Now().Add(1 * time.Minute),
+		// lastUnsuccessfulProbe:  time.Now().Add(2 * time.Minute),
```

Comment:
Do we not need this anymore?

Reply:
I commented it out so that the value is 0 or you can say never unsuccessful

Reply:
i added a comment also so now it's clear